### ISSUE TITLE
HTTP status code 410 error

### ISSUE THREAD

**Issue by beenotung** [skilion/onedrive#129]
Need to update the api address?
```
HTTP request returned status code 410 (Gone)
HTTP request returned status code 410 (Gone)
sync.SyncException@src/sync.d(106): HTTP request returned status code 410 (Gone)
----------------
??:? void sync.SyncEngine.applyDifferences() [0x59c7dc]
??:? void main.performSync(sync.SyncEngine) [0x58ca8b]
??:? _Dmain [0x58c2da]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFZv [0x5a9a6e]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5a99b8]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x5a9a2a]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5a99b8]
??:? _d_run_main [0x5a9929]
??:? main [0x58fa5d]
??:? __libc_start_main [0x7efb7290]
onedrive.OneDriveException@src/onedrive.d(348): HTTP request returned status code 410 (Gone)
----------------
??:? std.json.JSONValue onedrive.OneDriveApi.get(const(char)[]) [0x5940ea]
??:? void sync.SyncEngine.applyDifferences() [0x59c55f]
??:? void main.performSync(sync.SyncEngine) [0x58ca8b]
??:? _Dmain [0x58c2da]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFZv [0x5a9a6e]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5a99b8]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x5a9a2a]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5a99b8]
??:? _d_run_main [0x5a9929]
??:? main [0x58fa5d]
??:? __libc_start_main [0x7efb7290]
Segmentation fault (core dumped)
```

**Comment by sidhant** [comment]
Same here:
HTTP request returned status code 410 (Gone)
HTTP request returned status code 410 (Gone)
sync.SyncException@src/sync.d(106): HTTP request returned status code 410 (Gone)
----------------
??:? void sync.SyncEngine.applyDifferences() [0x59b8a4]
??:? void main.performSync(sync.SyncEngine) [0x58ba4b]
??:? _Dmain [0x58b28e]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFZv [0x5a8ade]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5a8a28]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x5a8a9a]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5a8a28]
??:? _d_run_main [0x5a8999]
??:? main [0x58ea95]
??:? __libc_start_main [0xc4be382f]
onedrive.OneDriveException@src/onedrive.d(348): HTTP request returned status code 410 (Gone)
----------------
??:? std.json.JSONValue onedrive.OneDriveApi.get(const(char)[]) [0x593122]
??:? void sync.SyncEngine.applyDifferences() [0x59b627]
??:? void main.performSync(sync.SyncEngine) [0x58ba4b]
??:? _Dmain [0x58b28e]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFZv [0x5a8ade]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5a8a28]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x5a8a9a]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5a8a28]
??:? _d_run_main [0x5a8999]
??:? main [0x58ea95]
??:? __libc_start_main [0xc4be382f]

**Comment by Simon-Tang** [comment]
@sidhant do you have different output than @beenotung does, or did you just paste the exact same output (minus the last line)?

**Comment by sidhant** [comment]
@Simon-Tang I posted error from my logs.

**Comment by skilion** [comment]
Solved in 7257c4c9bfb07efe4e333f562868e2ad80b739d3

**Comment by Simon-Tang** [comment]
Should note that it looks like you have to call

`onedrive --resync`

or you get this:

```
Status token expired, resyncing
std.json.JSONException@std/json.d(248): JSONValue is not an object
----------------
??:? pure @safe void std.exception.bailOut!(std.json.JSONException).bailOut(immutable(char)[], ulong, const(char[])) [0x5fe34a]
??:? pure @safe bool std.exception.enforce!(std.json.JSONException, bool).enforce(bool, lazy const(char)[], immutable(char)[], ulong) [0x5fe2cd]
??:? std.json.JSONValue.objectinout(pure ref @property inout(std.json.JSONValue[immutable(char)[]]) function()) [0x5c4701]
??:? void sync.SyncEngine.applyDifferences() [0x5a1822]
??:? void main.performSync(sync.SyncEngine) [0x5919a7]
??:? _Dmain [0x5911f6]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFZv [0x5aed6e]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5aecb8]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x5aed2a]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x5aecb8]
??:? _d_run_main [0x5aec29]
??:? main [0x594979]
??:? __libc_start_main [0x83bf8f44]
Segmentation fault
```

**Comment by skilion** [comment]
Thanks, fixed in e33f566ecf19c68e5d5b70775aaa17f9fefe253e

**Comment by sidhant** [comment]
Thanks for the quick fix!!
